packages/react-reconciler/src/ReactFiberWorkLoop.new.js

```diff
@@ -710,33 +712,68 @@ function ensureRootIsScheduled(root: FiberRoot, currentTime: number) {
   // This returns the priority level computed during the `getNextLanes` call.
   const newCallbackPriority = returnNextLanesPriority();
 
-  if (nextLanes === NoLanes) {
-    // Special case: There's nothing to work on.
-    if (existingCallbackNode !== null) {
-      cancelCallback(existingCallbackNode);
+  if (enableDiscreteEventMicroTasks) {
+    if (nextLanes === NoLanes) {
+      // Special case: There's nothing to work on.
+      if (existingCallbackNode !== null) {
+        cancelCallback(existingCallbackNode);
+      }
       root.callbackNode = null;
       root.callbackPriority = NoLanePriority;
+      return;
     }
-    return;
-  }
 
-  // Check if there's an existing task. We may be able to reuse it.
-  if (existingCallbackNode !== null) {
+    // Check if there's an existing task. We may be able to reuse it.
     const existingCallbackPriority = root.callbackPriority;
-    if (existingCallbackPriority === newCallbackPriority) {
-      // The priority hasn't changed. We can reuse the existing task. Exit.
+    if (
+      existingCallbackPriority !== NoLanePriority &&
+      existingCallbackPriority !== InputDiscreteLanePriority
```

Comment:
I just noticed this check, why is this necessary? Checking `existingCallbackNode` is sufficient. If you want the invariant I would make that a DEV-only warning so we don't do an extra check in prod.

Reply:
I thought the goal was to not check existingCallbackNode and specify the priorities that shouldn't have a callback scheduled?